sdk/src/asset_handlers/woff_io.rs

```diff
@@ -0,0 +1,1739 @@
+
+impl std::fmt::Debug for TableTag {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}{}{}{}", self.data[0], self.data[1], self.data[2], self.data[3])
+    }
+}
+
+/// Actual tag names for real font tables in which we take an interest.
+///
+/// Tag for the 'C2PA' table.
+const C2PA_TABLE_TAG: TableTag = TableTag { data: *b"C2PA", };
+/// Tag for the 'head' table in a font.
+const HEAD_TABLE_TAG: TableTag = TableTag { data: *b"head", };
+
+/// Ad-hoc "tag" names for font-file entities which are not true tables, but
+/// which we'd like to treat as tables.
+///
+/// These fake tag names:
+/// 
+/// 1. **Must** be extremely unlikely to ever appear as real tables in any input
+///    font, since they will conflict. Therefore, they should probably contain
+///    unprintable ASCII values like 0x00, 0x7F, since most fonts specify that
+///    tables should have 4-character ASCII names, padded with spaces if needed.
+/// 
+/// 2. **May** need to be chosen to sort into the correct physical order when
+///    enumerated with all other tables and non-tables in the font. Whether this
+///    matters depends on the implementation details.
+
+/// WOFF 1.0 WOFFHeader
+const WOFF_HEADER_TAG: TableTag = TableTag { data: *b"\x00\x00\x00w", };   // Then SFNT header could be "000s", and all the "header" chunks will BTreeMap to start-of-file.
+/// WOFF 1.0 / 2.0 trailing XML metadata
+const WOFF_METADATA_TAG: TableTag = TableTag { data: *b"\x7F\x7F\x7FM", }; // Needs to sort to (almost the) end.
+/// WOFF 1.0 / 2.0 trailing private data
+const WOFF_PRIVATE_DATA_TAG: TableTag = TableTag { data: *b"\x7F\x7F\x7FP", }; // Needs to sort to (almost the) end.
+/// Pseudo-tag for the table directory.
+const TABLE_DIRECTORY_TAG: TableTag = TableTag { data: *b"\x00\x00\x01D", }; // Sorts to just-after HEADER tag.
+
+/// 32-bit font-format identifier.
+///
+/// Embedded OpenType and MicroType Express formats cannot be detected with a
+/// simple magic-number sniff. Conceivably, EOT could be dealt with as a
+/// variation on SFNT, but MTX needs more exotic handling.
+enum Magic {
+    /// 'OTTO' - OpenType
+    OpenType = 0x4f54544f,
+    /// FIXED 1.0 - TrueType (or possibly v1.0 Embedded OpenType)
+    TrueType = 0x00010000,
+    /// 'wOFF' - WOFF 1.0
+    Woff = 0x774f4646,
+    /// 'wOF2' - WOFF 2.0
+    Woff2 = 0x774f4632,
+}
+
+/// Used to attempt conversion from u32 to a Magic value.
+impl TryFrom<u32> for Magic {
+    type Error = crate::error::Error;
+
+    /// Tries to convert from u32 to a valid font version.
+    fn try_from(v: u32) -> core::result::Result<Self, Self::Error> {
+        match v {
+            tt if tt == Magic::TrueType as u32 => Ok(Magic::TrueType),
+            ot if ot == Magic::OpenType as u32 => Ok(Magic::OpenType),
+            w1 if w1 == Magic::Woff as u32 => Ok(Magic::Woff),
+            w2 if w2 == Magic::Woff2 as u32 => Ok(Magic::Woff2),
+            _unknown => Err(Error::FontLoadError),
```

Comment:
This really is a different error than used, I think a new error type would help inform exactly what happened here. Maybe something like:

```rust
#[error("Unknown font magic")]
UnknownFontMagic
```